### PR TITLE
WIP: Convert updates

### PR DESCRIPTION
**Pull request by abbiemery**
Some adjustments to increase the scope of the converts.

To do:
ADCore: 
- Make manual changes to the pluginSrc to include paramSets etc for those on the same level as NDFileHDF5
- Input manual fields for NDFIleHDF5.pvi.producer.yaml

PVI:
- Need to look at pv grouping for the edm screens.

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -5,6 +5,8 @@ verify_ssl = true
 
 [dev-packages]
 pvi = {editable = true, extras = ["dev"], path = "."}
+mypy = "*"
+flake8 = "*"
```
Comment (GDYendell) [inline Pipfile:9]: These should be in setup.cfg

Comment (GDYendell) [inline Pipfile:9]: These should be in setup.cfg

File: src/pvi/__main__.py [hunk 2]
```diff
@@ -131,10 +146,15 @@ def asyn(
...
-        if isinstance(parameter, AsynParameter):
-            parameter.index_name = index_map[parameter.get_drv_info()]
+    if source_converter.define_strings:
+        index_map = source_converter.get_info_index_map()
+        for parameter in walk(producer.parameters):
+            if isinstance(parameter, AsynParameter):
+                try:
+                    parameter.index_name = index_map[parameter.get_drv_info()]
+                except KeyError:
+                    parameter.index_name = "None"
```
Comment (GDYendell) [inline src/pvi/__main__.py:156]:
```suggestion
                    parameter.index_name = index_map.get(parameter.get_drv_info(), "None")
```

File: src/pvi/_convert/_source_convert.py [hunk 3]
```diff
@@ -207,11 +207,24 @@ def _convert_h(self, original_h: str) -> str:
...
+            h_text = h_text.replace(
+                protected_str,
+                protected_str + "\n" + param_set_definition,
+                1,
+            )
+        else:
+            h_text = h_text.replace(
+                class_str,
+                class_str + "\n" + "protected:" + "\n" + param_set_definition,
+            )
```
Comment (GDYendell) [inline src/pvi/_convert/_source_convert.py:227]: Could do with a brief comment for each case to say e.g. Add param set to existing protected member list and Create protected member list with param set

File: src/pvi/__main__.py (hunk 2 above)
Comment (abbiemery) [reply src/pvi/__main__.py]: I fixed this in one of the commits.

**Pull request merged**